src/Service/EncryptionService.php

```diff
@@ -56,9 +75,15 @@ public function encryptField($value): EncryptedColumnVO
     {
         if ($value instanceof LazyLoadingInterface) {
             /** @var LazyLoadingInterface|ValueHolderInterface $value */
+            // If the value hasn't been decrypted; it hasn't been changed. Don't bother reencrypting unless it
+            // was encrypted using a different configuration
             if (!$value->isProxyInitialized()) {
-                //if data hasn't been encrypted, we don't need to change it; set it back to what it was at load
-                return $this->originalValues[spl_object_hash($value)];
+                $original = $this->originalValues[spl_object_hash($value)];
+                if (
+                    !$original->needsReencryption($this->encryptor->getIdentifier(), $this->serializer->getIdentifier())
```

Comment:
Been considering this but given that one of the goals of this library is to stop non-crypto devs from needing to know what the best options are for encrypting securely - the only parameter available for them to change on an encryptor will be it's key, which I plan to handle separately.

Any change in crypto params made by this library need to be backward compatible so either a change in parameters becomes a new encryptor (including both in the lib to enable existing data to be read) or the encryptor itself needs to encode (in the data) what the parameters used were.

In the event that this isn't sufficient the identifier is now a VO concerned with it's own equality, so it could be extended in future to contain a version if required.